test/no-mutating-assign.js

```diff
@@ -23,6 +23,8 @@ ruleTester.run('no-mutating-assign', rule, {
     'Object.foo(a, b);',
     'Object.assign({});',
     'Object.assign({}, b);',
+    'Object.assign(Object.create(null));',
```

Comment:
Could you add the following tests?
Valid:
- 'Object.assign(Object.create(a))' // Any call of Object.create should probably be allowed

Invalid:
- 'Object.assign(Object.create)' // We don't want to mutate the function

Reply:
Ah, you're right. Scrap the valid test I proposed then

Reply:
Pushed up the invalid test case, @jfmengels 👍